url4e79c36700_JacenLakiir_Tivoo_tgz-pJ8-MainJ8/src/tivoo/output/HTMLBuilder.java

```diff
@@ -0,0 +1,382 @@
+    {
+        String[] days =
+            new String[] {
+                    "Sunday",
+                    "Monday",
+                    "Tuesday",
+                    "Wednesday",
+                    "Thursday",
+                    "Friday",
+                    "Saturday" };
+        List<String> dayList = new ArrayList<String>();
+        for (String d : days)
+            dayList.add(d);
+        return dayList;
+    }
+    
+    private static List<String> initializeMonthsList ()
+    {
+        String[] months =
+            new String[] {
+                    "January",
+                    "February",
+                    "March",
+                    "April",
+                    "May",
+                    "June",
+                    "July",
+                    "August",
+                    "September",
+                    "October",
+                    "November",
+                    "December" };
+        List<String> monthList = new ArrayList<String>();
+        for (String m : months)
+            monthList.add(m);
+        return monthList;
+    }
+
+    private String determineDetailPageDirectory (String summaryPageFileName)
+    {
+        StringBuilder detailPageDirectory = new StringBuilder();
+        if (summaryPageFileName.contains("/"))
+        {
+            int lastFolder = summaryPageFileName.lastIndexOf("/");
+            detailPageDirectory.append(summaryPageFileName.substring(0, lastFolder + 1));
+            detailPageDirectory.append(DETAIL_PAGE_FOLDER);
+        }
+        else
+        {
+            detailPageDirectory.append(DETAIL_PAGE_FOLDER);
+        }
+        return detailPageDirectory.toString();
+    }
+    
+    private void buildSummaryPage (List<Event> eventList) throws IOException
+    {
+        if (mySummaryPageFileName.contains("/"))
+        {
+            File outputDirectory = new File(myDetailPageDirectory);
+            outputDirectory.mkdirs();
+        }
+        File summaryPage = new File(mySummaryPageFileName);
+        summaryPage.createNewFile();
+        writeSummaryPageHTML(summaryPage, eventList);
+    }
+    
+    private void buildDetailPages (List<Event> eventList) throws IOException
+    {
+        File detailsDirectory = new File(myDetailPageDirectory);
+        detailsDirectory.mkdirs();
+        for (Event currentEvent : eventList)
+        {
+            String detailPageURL = createDetailsPageURL(currentEvent);
+            File detailPage = new File(myDetailPageDirectory + "/" + detailPageURL);
+            boolean isNewFile = detailPage.createNewFile();
+            if (isNewFile)
+                writeDetailsPageHTML(currentEvent, detailPage);
+        }
+    }
+    
+    private void writeSummaryPageHTML (File summaryPage, List<Event> eventList) throws IOException
+    {
+        FileOutputStream fos = new FileOutputStream(summaryPage);
+        OutputStreamWriter out = new OutputStreamWriter(fos, "UTF-8");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *out*. The resource is referenced at lines 234, 248. The resource is closed at line 249. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).